src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -1468,9 +1480,25 @@ private void TrackNullableStateForAssignment(BoundNode node, Symbol assignmentTa
 
                 if (targetType.IsReferenceType)
                 {
+                    bool? targetIsNullable;
+                    if (slot > 0)
+                    {
+                        var identifierAndNullability = variableBySlot[slot];
+                        if (inferNullability)
+                        {
+                            identifierAndNullability = new VariableIdentifierAndNullability(identifierAndNullability.Identifier, !valueIsNotNull);
+                            variableBySlot[slot] = identifierAndNullability;
+                        }
```

Comment:
Perhaps I don't understand the design here. variableBySlot was not designed to be mutated as the analysis proceeds. It was intended to be set once per slot, at the time the slot is allocated. Nullability should be part of the state since it changes during analysis and from place to place in the program.

Reply:
The intent here was not to mutate the nullability but simply to set it once, at the point the local is initialized. But @AlekseyTs gave the same feedback that the nullability from the initializer should be part of the state. I'll investigate that.